alf/config_helpers.py

```diff
@@ -94,19 +99,71 @@ def get_action_spec():
     return env.action_spec()
 
 
+_is_parsing = False
+
+
+def parse_config(conf_file, conf_params):
+    """Parse config file and config parameters
+
+    Note: a global environment will be created (which can be obtained by
+    alf.get_env()) and random seed will be initialized by this function using
+    common.set_random_seed().
+
+    Args:
+        conf_file (str): The full path of the config file.
+        conf_params (list[str]): the list of config parameters. Each one has a
+            format of CONFIG_NAME=VALUE.
+    """
+    global _is_parsing
+    _is_parsing = True
+
+    try:
+        if conf_params:
+            for conf_param in conf_params:
+                pos = conf_param.find('=')
+                if pos == -1:
+                    raise ValueError("conf_param should have a format of "
+                                     "'CONFIG_NAME=VALUE': %s" % conf_param)
+                config_name = conf_param[:pos]
+                config_value = conf_param[pos + 1:]
+                config_value = eval(config_value)
+                config1(config_name, config_value, mutable=False)
+
+        runpy.run_path(conf_file)
+    finally:
+        _is_parsing = False
+
+    # Create the global environment and initialize random seed
+    get_env()
+
+
 def get_env():
     """Get the global training environment.
 
     Note: you need to finish all the config for environments and
     TrainerConfig.random_seed before using this function.
 
+    Note: random seed will be initialized in this function.
+
     Returns:
         AlfEnvironment
     """
     global _env
     if _env is None:
-        trainer_config = TrainerConfig(root_dir='')
-        _env = create_environment(seed=trainer_config.random_seed)
+        if _is_parsing:
+            random_seed = get_config_value('TrainerConfig.random_seed')
+        else:
+            # We construct a TrainerConfig object here so that the value
+            # configured through gin-config can be properly retrieved.
+            train_config = TrainerConfig(root_dir='')
```

Comment:
I'm confused.  This config_helper.py is only used by alf config, not gin-config?  Why do we need to care about gin-config here?